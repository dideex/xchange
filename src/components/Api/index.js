import {noty} from '../common'

export class Api {
  constructor() {
    this.backend = 'http://localhost:3030'
    this.isNetworkError = false
  }

  post(endpoint, data = {}, token = null) {
    return fetch(`${this.backend}/api/${endpoint}`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
      body: JSON.stringify(data),
    }).then(response => response.json())
  }

  get(endpoint, params = '', token = null) {
    return fetch(`${this.backend}/api/${endpoint}${params}`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
    }).then(response => response.json())
  }
  
  errorEmitter = fn => ({err, errCode, ...data}) => {
    if (err) {
      this.isNetworkError = true
      console.error('errorEmitter ___ ', errCode)
      noty(err, 'error')
    } else fn(data)
  }
}

export default new Api()
