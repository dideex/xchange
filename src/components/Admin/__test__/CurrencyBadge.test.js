import React from 'react'
import Component from '../CurrencyBadge'
import {shallow, mount} from 'enzyme'
import Api from '../../../components/Api'
// import Common from '../../common'

jest.mock('../../../components/Api', () => ({
  post: () => {},
  get: () => {},
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

jest.mock('../../../components/common', () => ({
  noty: () => {},
}))

const fakeData = {name: 'Test name', price_usd: 'Test usd price', token: 'fakeToken'}

describe('Tests', () => {
  describe('Markup', () => {
    it('Base markup', () => {
      const data = {}
      const wrapper = mount(<Component data={data} />)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Data markup', () => {
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Data markup opened', () => {
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      wrapper.simulate('click')

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('State behavriour', () => {
    it('Should open after click', () => {
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      wrapper.simulate('click')

      expect(wrapper.state().show).toBeTruthy()
    })

    it('Reserved should change', () => {
      const fakeValue = 'Test value'
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      wrapper.simulate('click')

      wrapper
        .find('input[name="reserved"]')
        .simulate('change', {target: {value: fakeValue, name: 'reserved'}})
      expect(wrapper.state().reserve).toBe(fakeValue)
    })

    it('Source should change', () => {
      const fakeValue = 'Test value'
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      wrapper.simulate('click')

      wrapper
        .find('input[name="source"]')
        .simulate('change', {target: {value: fakeValue, name: 'source'}})
      expect(wrapper.state().source).toBe(fakeValue)
    })

    it('Minimal should change', () => {
      const fakeValue = 'Test value'
      const data = fakeData
      const wrapper = mount(<Component data={data} />)
      wrapper.simulate('click')

      wrapper
        .find('input[name="minimal"]')
        .simulate('change', {target: {value: fakeValue, name: 'minimal'}})
      expect(wrapper.state().minimal).toBe(fakeValue)
    })
  })

  describe('Post behaviour', () => {
    it('Post hanlder should be invoked by button click', () => {
      const wrapper = mount(<Component data={fakeData} />)
      wrapper.instance()._pushSettings = jest.fn()
      wrapper.simulate('click')
      wrapper.find('button').simulate('click')

      expect(wrapper.instance()._pushSettings).toHaveBeenCalledTimes(1)
    })

    it.only('Post hanlder should be invoked by button click', async () => {
      Api.post = jest.fn(() => Promise.resolve())
      const wrapper = mount(<Component data={fakeData} />)
      wrapper.simulate('click')
      // await wrapper.find('button').simulate('click')
      await wrapper.instance()._pushSettings()

      expect(Api.post).toHaveBeenCalledTimes(1)
    })
  })
})
