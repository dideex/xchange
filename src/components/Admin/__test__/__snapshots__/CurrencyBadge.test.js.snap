// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Markup Base markup 1`] = `"<div class=\\"css-72ota7\\"></div>"`;

exports[`Tests Markup Data markup 1`] = `"<div class=\\"css-72ota7\\">Test name</div>"`;

exports[`Tests Markup Data markup opened 1`] = `"<div class=\\"css-72ota7\\">Test name<ul><li>Цена в usd Test usd price</li><li><p>Резервы:</p><input name=\\"reserved\\" type=\\"text\\" value=\\"\\"></li><li><p>Кошелек для перевода:</p><input name=\\"source\\" type=\\"text\\" value=\\"\\"></li><li><p>Минимальная сумма для перевода:</p><input name=\\"minimal\\" type=\\"text\\" value=\\"\\"></li><button>Сохранить</button></ul></div>"`;
